airbyte_cdk/sources/declarative/incremental/concurrent_partition_cursor.py

```diff
@@ -141,21 +145,16 @@ def close_partition(self, partition: Partition) -> None:
             raise ValueError("stream_slice cannot be None")
 
         partition_key = self._to_partition_key(stream_slice.partition)
```

Comment:
nit and outside of scope: I like the fact that we compute this value only once so that performance should be better. Should we do the same on `_generate_slices_from_partition`?

Reply:
I see this change as part of https://github.com/airbytehq/airbyte-python-cdk/pull/343/files#diff-40bb6219c9d2c70a591e9e3b2209087e1b01eb44bfe36c15ff22b43506ac69faR239 so I'm going to resolve this here